test/suites/optional/observability.robot

```diff
@@ -0,0 +1,63 @@
+*** Settings ***
+Documentation       This suite performs basic log scans to determine if the opentelemetry-collector is exporting data
+...                 and in a healthy state
+
+Library             OperatingSystem
+Library             String
+Library             ../../resources/journalctl.py
+Resource            ../../resources/kubeconfig.resource
+Resource            ../../resources/common.resource
+
+Suite Setup         Setup Suite And Set Journal Cursor
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${JOURNAL_CUR}      ${EMPTY}
+
+
+*** Test Cases ***
+Logs Should Contain Exported Metric Data
+    [Documentation]    OpenTelemetry logs should contain the data from the LogsExporter and MetricsExporter
+
+    ${pattern}    Catenate    SEPARATOR=
+    ...    info\\s+MetricsExporter\\s+{"kind": "exporter", "data_type": "metrics", "name": "debug", "resource metrics":
+    ...    \ [0-9]+, "metrics": [0-9]+, "data points": [0-9]+}
+    Pattern Should Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+Logs Should Contain Exported Log Data
+    [Documentation]    OpenTelemetry logs should not contain errors by any component (reciever, processor,exporter).
+
+    ${pattern}    Catenate
+    ...    SEPARATOR=
+    ...    info\\s+LogsExporter\\s+\\{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": [0-9]+,
+    ...    \ "log records": [0-9]+\\}
+    Pattern Should Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+Logs Should Not Contain Reciever Errors
+    [Documentation]    Internal reciever errors are not treated as fatal. Typically these are due to a miconfiguration
+    ...    and thus indicate the provided default config should be reviewed.
+
+    ${pattern}    Catenate    SEPARATOR= \\s+\\{"error":.*\\}
+    Pattern Should Not Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+
+*** Keywords ***
+Setup Suite And Set Journal Cursor
+    [Documentation]    The service starts after MicroShift starts and thus will start generating pertinant log data
+    ...    right away. When the suite is executed, immediately get the cursor for the microshift-observability unit.
+    Setup Suite
+    ${cur}    Get Journal Cursor    unit=microshift-observability
+    Set Suite Variable    ${JOURNAL_CUR}    ${cur}
+    Wait Until Keyword Succeeds    1 min    5 sec
```

Comment:
We may want to increase the wait interval to say 5m?

Reply:
If we've made it to this point, then the observability service is running and generating output. This check is more of a precaution to ensure the test is only checking output that's been generated very recently. Otherwise we may pickup errors or failures caused by service restarts or reboots. That can happen for instance if another test set is run before the observability tests.